### PR TITLE
[GBP: No update] Golem hands fix, probably

### PR DESCRIPTION
**Pull request by ElorgRHG**
## What Does This PR Do

Fixes golems that are not adamantine to have hands that are.

But I'll say this right at the start, this is not the greatest solution if admemery gets involved, but since golems should not be able to have prosthetics limbs under normal circumstances since they cannot lose their limbs at all + you cannot augment them, this should be fine?

Could only think of this solution since golems have a variable that decides what color their body will be, recoloring their icobase, which is adamantine by default. Using anything other than icobase wouldn't work as far as I know?

## Why It's Good For The Game

Fix :)

## Images of changes

before: 
image

after:
image

## Testing

Spawned a golem before the fix, and after
Arranged images in `Images of changes`

## Changelog
:cl:
fix: Fixed all golems having adamantine hands.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 1]
```diff
@@ -418,6 +418,15 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
 	var/obj/item/organ/external/hand/l_hand = get_limb_by_name("l_hand")
 	var/obj/item/organ/external/hand/right/r_hand = get_limb_by_name("r_hand")
 
+	if(isgolem(src))
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/human/update_icons.dm:421]: scope this to 
`/mob/living/carbon/human/golem/update_hands_layer()`

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 2]
```diff
@@ -418,6 +418,15 @@ GLOBAL_LIST_EMPTY(damage_icon_parts)
...
+		var/datum/species/golem/G = src.dna.species
```
Comment (Sirryan2002) [inline code/modules/mob/living/carbon/human/update_icons.dm:422]:
```suggestion
		var/datum/species/golem/G = dna.species
```
remember that you don't need `src` to refer to variables and methods on the object you're currently scoped to, src is already implied
`src.proc` is the same thing as `proc` in almost every case (except constructors but this isn't one of those)

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/update_icons.dm (hunk 1 above)
Comment (ElorgRHG) [reply code/modules/mob/living/carbon/human/update_icons.dm]: Scoped

_I think_

**Review by S34NW — APPROVED** [review]

**Review by farie82 — APPROVED** [review]
Quite some dupe code. But I can't think of a quick way to do it in a much cleaner manner

**Comment by Sirryan2002** [comment]
Yeah I agree with the dupe code, in the future, the code should be designed to to allow overrides that call parent, that would probably involved some `post-update` proc

**Pull request merged**